### PR TITLE
Fix LocaleCookieRedirect middleware

### PR DESCRIPTION
**Pull request by jochensengier**
I've been using this package for quite a few years and only recently been having some weird issues:

When the cookie locale contains an empty string, the following exception occurs:
```
Arcanedev\Localization\Exceptions\UnsupportedLocaleException
Locale '' is not in the list of supported locales. 
```
I'm still unsure why the locale from the cookie would be empty, but I believe we can fix this by checking if the locale from the cookie is actually supported, just like the segment is checked (line 42).

This will also prevent an exception if a user has a locale set in their cookie which is no longer supported.
I've run into this situation recently and the only fix was to force remove the locale cookie on each request.

Sometimes the first segment of the URL is stored as the locale in the cookie, throwing a similar exception:
```
Arcanedev\Localization\Exceptions\UnsupportedLocaleException
Locale 'products' is not in the list of supported locales. 
```
I might be off, but this probably happens because the segment is stored instead of the locale from the cookie (line 44).

### CODE REVIEW

File: src/Middleware/LocaleCookieRedirect.php [hunk 1]
```diff
@@ -39,9 +39,9 @@ public function handle(Request $request, Closure $next)
 
         $locale  = $request->cookie('locale', null);
 
-        if ($locale !== null && ! $this->isDefaultLocaleHidden($locale)) {
+        if ($this->localization->isLocaleSupported($locale) && ! $this->isDefaultLocaleHidden($locale)) {
```
Comment (arcanedev-maroc) [inline src/Middleware/LocaleCookieRedirect.php]: Just revert the changes and add a check to verify the locale is not empty instead

Comment (jochensengier) [reply src/Middleware/LocaleCookieRedirect.php]: But then it will still throw an exception if a user has a locale in their cookie which is no longer supported?

Comment (arcanedev-maroc) [reply src/Middleware/LocaleCookieRedirect.php]: Can you write a test to reproduce the issue ?

Comment (jochensengier) [reply src/Middleware/LocaleCookieRedirect.php]: Please triple check and have mercy, I'm not an expert at writing tests 🙈
It only passes with the code changes from my PR.

**Comment by arcanedev-maroc** [comment]
I have mixed feelings about this PR. It changes two things:

#### First issue: **if the locale is an empty string**

I'm ok with this because it can happen if you're using a other package or your app that uses the same name for the session `locale`. 

**Solution: Updating the LocaleCookieRedirect check**
```php
// We change this check
if ($locale !== null && ...)

// Into this
if ( ! empty($locale) && ...)
```

#### Second Issue: The `Arcanedev\Localization\Exceptions\UnsupportedLocaleException` exception

There are users (including me) expecting to have this exception to do other logic in their projects. 

This is not a bug, This **issue** was caused by a **decision** of dropping a supported locale. Now your visitors has a piece of data that needs to be cleaned.

The solution is pretty easy to implement: **The error handler**

Check the [official documentation](https://laravel.com/docs/errors) about it. 

And this is an example why i don't want to change this behaviour:
```php
<?php

namespace App\Exceptions;

use Arcanedev\Localization\Exceptions\UnsupportedLocaleException;
use Illuminate\Foundation\Exceptions\Handler as ExceptionHandler;

class Handler extends ExceptionHandler
{
    /**
     * A list of the exception types that are not reported.
     *
     * @var array
     */
    protected $dontReport = [
        //
    ];

    /**
     * A list of the inputs that are never flashed for validation exceptions.
     *
     * @var array
     */
    protected $dontFlash = [
        'password',
        'password_confirmation',
    ];

    /**
     * Register the exception handling callbacks for the application.
     *
     * @return void
     */
    public function register()
    {
        $this->renderable(function (UnsupportedLocaleException $e) {
            // Clean the outdated data
            session()->forget('locale');

            // Save this 'side effect' somewhere and other stuff...

            return redirect()->to(config('app.locale')); // Or redirect to 'select your language' page for example...
        });
    }
}
```

Keep in mind, having an exception is not a bad thing in laravel :smile:

**Comment by jochensengier** [comment]
I completely agree with your comments and suggestions.
Thanks for taking the time to review my PR and the tip about the error handler!

File: src/Middleware/LocaleCookieRedirect.php [hunk 2]
```diff
@@ -39,9 +39,9 @@ public function handle(Request $request, Closure $next)
 
         $locale  = $request->cookie('locale', null);
 
-        if ($locale !== null && ! $this->isDefaultLocaleHidden($locale)) {
+        if (! empty($locale) && ! $this->isDefaultLocaleHidden($locale)) {
             if ( ! is_null($redirect = $this->getLocalizedRedirect($locale)))
-                return $redirect->withCookie(cookie()->forever('locale', $segment));
```
Comment (arcanedev-maroc) [inline src/Middleware/LocaleCookieRedirect.php:44]: Can you revert this line?

**Review by arcanedev-maroc — CHANGES_REQUESTED** [review]

File: src/Middleware/LocaleCookieRedirect.php (hunk 2 above)
Comment (arcanedev-maroc) [reply src/Middleware/LocaleCookieRedirect.php:44]: Never mind

**Review by arcanedev-maroc — APPROVED** [review]
